src/EventTickets/resources/admin/components/EventFormModal/index.tsx

```diff
@@ -0,0 +1,157 @@
+import {SubmitHandler, useForm} from 'react-hook-form';
+import {__} from '@wordpress/i18n';
+import styles from './EventFormModal.module.scss';
+import FormModal from '../FormModal';
+import EventTicketsApi from '../api';
+
+/**
+ * Get the next sharp hour
+ *
+ * @unreleased
+ */
+const getNextSharpHour = (hoursToAdd: number) => {
+    const date = new Date();
+    date.setHours(date.getHours() + hoursToAdd, 0, 0, 0);
+
+    return date;
+};
+
+/**
+ * Format a given date to be used in datetime inputs
+ *
+ * @unreleased
+ */
+const getDateString = (date: Date) => {
+    if (!date) {
+        return;
+    }
+
+    const offsetInMs = date.getTimezoneOffset() * 60 * 1000;
+    const dateWithOffset = new Date(date.getTime() - offsetInMs);
+
+    return dateWithOffset.toISOString().slice(0, -8);
+};
+
+/**
+ * Event Form Modal component
+ *
+ * @unreleased
+ */
+export default function EventFormModal({isOpen, handleClose, apiSettings, title, event}: EventModalProps) {
+    const API = new EventTicketsApi(apiSettings);
```

Comment:
Instead of passing in the `apiSettings` to the component, should we be passing in the `EventTicketsApi` object itself?

Reply:
This component is shared between Events ListTable and Event Details page. Each has distinct apiSettings being passed so I thought to keep each of them managing those. For instance, in the list table we have the `GiveEventTickets` object while in the details page we have the `GiveEventTicketsDetails`.